frontend/app/(tabs)/home.tsx

```diff
@@ -72,14 +62,42 @@ const Index = () => {
     } finally {
       setIsLoading(false);
     }
-  }, [page, isLoading]);
+  }, [page, isLoading, searchText]);
+
+  const handleSearch = (text: string) => {
+    setSearchText(text);
+
+    if (searchTimeout.current) {
+      clearTimeout(searchTimeout.current);
+    }
+
+    searchTimeout.current = setTimeout(() => {
+      if (text.trim().length >= 2) {
+        loadInitialCards(text);
```

Comment:
let's call this method `fetchCards`

Reply:
let's add a variable for `text.trim()` and use it here and below, we should also only send the trimmed text to the backend